### PR TITLE
Ev/charges and payments

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
Hi to all 

This branch purpose is to build a history table for the Administration->codes component. 
This feature can be disabled from Globals->Billing->Save prices history
The history table provides a more readable way to track changes
in the codes then the log table . 
It is also sets the ground for reports on the codes in general and prices in specifics. 

It does also include a minor change in the Administration->codes modifier field size. 

We post this change since our team needs it and hopefully other will find it useful too  :) 

Thanks ,Eyal

### CODE REVIEW

File: interface/patient_file/encounter/superbill_custom_full.php [hunk 1]
```diff
@@ -153,6 +154,45 @@ function bucks($amount) {
...
+      $fee=json_encode($_POST['fee']);
+      $code_sql= sqlFetchArray(sqlStatement("SELECT (ct_label) FROM code_types WHERE ct_id=".$code_type.";"));
+      $code_name='';
+
+      if ($code_sql){
+          $code_name=$code_sql['ct_label'];
+      }
+
+      $categorey_id= $_POST['form_superbill'];
+      $categorey_sql=sqlFetchArray(sqlStatement("SELECT (title ) FROM list_options WHERE list_id='superbill' AND option_id=".$categorey_id.";"));;
```
Comment (bradymiller) [inline interface/patient_file/encounter/superbill_custom_full.php]: use sql binding

**Comment by bradymiller** [comment]
This appears to log all code changes(not just pricing codes but also diagnostic etc.), so what are your thoughts on a more generic name for it since. Maybe something like codes_history or something like that. Also curious if you have any reports (currently or planned) to display this data?

File: interface/patient_file/encounter/superbill_custom_full.php [hunk 2]
```diff
@@ -153,6 +154,45 @@ function bucks($amount) {
...
+      $code_text  = $_POST['code_text'];
+      $modifier   = $_POST['modifier'];
+      $superbill  = $_POST['form_superbill'];
+      $related_code = $_POST['related_code'];
+      $cyp_factor = $_POST['cyp_factor'] + 0;
+      $active     = empty($_POST['active']) ? 0 : 1;
+      $reportable = empty($_POST['reportable']) ? 0 : 1; // dx reporting
+      $financial_reporting = empty($_POST['financial_reporting']) ? 0 : 1; // financial service reporting
+      $fee=json_encode($_POST['fee']);
+      $code_sql= sqlFetchArray(sqlStatement("SELECT (ct_label) FROM code_types WHERE ct_id=".$code_type.";"));
```
Comment (eyalvo-matrix) [inline interface/patient_file/encounter/superbill_custom_full.php]: Ok, thanks for the advice :)

File: interface/patient_file/encounter/superbill_custom_full.php (hunk 1 above)
Comment (eyalvo-matrix) [inline interface/patient_file/encounter/superbill_custom_full.php]: Ok

**Comment by eyalvo-matrix** [comment]
Hi brady I have updated the history names to match code history

**Comment by eyalvo-matrix** [comment]
Hi Brady is there anything else you want me to change ? 
Thanks

**Comment by bradymiller** [comment]
@eyalvo-matrix ,
nope. looks good. will bring this into the codebase this evening :)
(Only change I am going to make is to turn it on by default)

**Comment by bradymiller** [comment]
just brought this into the codebase. thanks for the contribution! -brady

**Comment by eyalvo-matrix** [comment]
Thanks have a nice day

**Pull request closed without merging**